File: c/7.c
#include <stdio.h>
#include <math.h>

int prime(int n)
{
	int i;

	if (n % 2 == 0)
		return 0;

	for (i = 3; i <= sqrt(n); i += 2)
		if (n % i == 0)
			return 0;

	return 1;
}

int main(void)
{
	int i = 1, p = 1;

	while (i += 2) {
		p += prime(i);
		if (p == 10001)
			break;
	}

	printf("%d\n", i);
	return 0;
}

File: js/7.js
function prime(n) {
    if (n == 2)
        return true;

    if (n % 2 == 0)
        return false;

    for (m = 3; m <= Math.sqrt(n); m += 2)
        if (n % m == 0)
            return false;

    return true;
}

primes = [2];

for (i = 3; primes.length != 10001; i += 2)
    if (prime(i))
        primes.push(i);

console.log(primes[10000]);
